src/cq_cam/command.py

```diff
@@ -104,34 +164,39 @@ def xyz_gcode(self, start: cq.Vector, precision=3) -> (str, cq.Vector):
 
     @abstractmethod
     def to_gcode(
-        self, previous_command: Union[Command, None], start: cq.Vector
-    ) -> (str, cq.Vector):
+        self, previous_command: Union[MotionCommand, None], start: cq.Vector
+    ) -> tuple[str, cq.Vector]:
         """Output all the necessary G-Code required to perform the command"""
         pass
 
     @abstractmethod
     def to_ais_shape(
         self, start: cq.Vector, as_edges=False, alt_color=False
-    ) -> (AIS_Shape, cq.Vector):
+    ) -> tuple[AIS_Shape, cq.Vector]:
         pass
 
 
-class ReferencePosition(Command):
-    def to_gcode(
-        self, previous_command: Union[Command, None], start: cq.Vector
-    ) -> (str, cq.Vector):
-        raise RuntimeError("Reference position may not generate gcode")
+class ConfigCommand(Command, ABC):
```

Comment:
Spindle here needs to be unique between commands as it can be specific to an operation. Coolant's master data always lives in job and shouldn't change between operations

Reply:
I just didn't implement the interface of config command to have a reference to Job as I didn't want a different interface to MotionCommand.